# Fix for ISPC issue 1595 and related llvm bug 43857.
# Fixing crash in constant propagation pass when returning nested structs.
Index: lib/Transforms/IPO/IPConstantPropagation.cpp
===================================================================
--- lib/Transforms/IPO/IPConstantPropagation.cpp
+++ lib/Transforms/IPO/IPConstantPropagation.cpp
@@ -232,7 +232,7 @@ static bool PropagateConstantReturn(Function &F) {
       // Find the index of the retval to replace with
       int index = -1;
       if (ExtractValueInst *EV = dyn_cast<ExtractValueInst>(Ins))
-        if (EV->hasIndices())
+        if (EV->getNumIndices() == 1)
           index = *EV->idx_begin();
 
       // If this use uses a specific return value, and we have a replacement,
